compiler/rustc_monomorphize/src/partitioning.rs

```diff
@@ -1176,6 +1188,18 @@ fn collect_and_partition_mono_items(tcx: TyCtxt<'_>, (): ()) -> MonoItemPartitio
         })
         .collect();
 
+    let autodiff_mono_items: Vec<_> = items
+        .iter()
+        .filter_map(|item| match *item {
+            MonoItem::Fn(ref instance) => Some((item, instance)),
+            _ => None,
+        })
+        .collect();
+
+    let autodiff_items =
+        autodiff::find_autodiff_source_functions(tcx, &usage_map, autodiff_mono_items);
```

Comment:
gate this on autodiff being enabled

Reply:
hmm, but we want to enable autodiff soonish by default. 
Also if you look at find_autodiff_from_source_functions, it does allocate an empty vector, realize there is nothing to do, and then returns the empty vector. I assume therefore it's the second iter().filteR_map over the items, wdyt (?)
I am tempted to fuse those two iterations and do a perf run with that?

Reply:
turns out either fusing was enough. or the other change fixed it.